client/src/components/layouts/default.tsx

```diff
@@ -242,13 +247,23 @@ function DefaultLayout({
             />
           ) : null}
           <SignoutModal />
-          {isChallenge && !examInProgress && (
+          {isChallenge &&
+          !examInProgress &&
+          (!isMobileLayout ||
+            (isMobileLayout &&
+              (/project$/.test(block as string) ||
+                (superBlock != SuperBlocks.RespWebDesignNew &&
+                  superBlock != SuperBlocks.JsAlgoDataStructNew &&
+                  superBlock != SuperBlocks.SciCompPy) ||
+                !isMobileHeight))) ? (
```

Comment:
To make things a little less complicated lets make a separate variable for the superblock filtering. 

Try to create one for the others also, it seems a bit hard to understand what is going on.

Reply:
I definitely agree with you, so I will try to making them separate variables to make the logic understandable.